### PR TITLE
Add function to know if TTimer is running (within gSystem)

### PR DESCRIPTION
**Pull request by ferdymercury**

### CODE REVIEW

File: core/base/inc/TTimer.h [hunk 1]
```diff
@@ -80,6 +80,7 @@ class TTimer : public TSysEvtHandler {
    Bool_t         IsSync() const { return fSync; }
    Bool_t         IsAsync() const { return !fSync; }
    Bool_t         IsInterruptingSyscalls() const { return fIntSyscalls; }
+   Bool_t         IsRunning();///< Check if timer is running within gSystem (has been started and did not finish yet).
```
Comment (pcanal) [inline core/base/inc/TTimer.h:83]: To match the rest of the file, the documentation string for the function should be in the source file.

**Comment by pcanal** [comment]
@phsft-bot build

File: core/base/inc/TTimer.h (hunk 1 above)
Comment (ferdymercury) [reply core/base/inc/TTimer.h]: Done

**Comment by ferdymercury** [comment]
> Build failed on windows10/cxx14.
> Running on null:C:\build\workspace\root-pullrequests-build
> [See console output](https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/106671/console).

I can't check the error log (no permission). Let me know if I can fix the error somehow. Thanks!

**Comment by Axel-Naumann** [comment]
@phsft-bot build

File: core/base/src/TTimer.cxx [hunk 2]
```diff
@@ -269,3 +269,14 @@ void TTimer::SingleShot(Int_t milliSec, const char *receiver_class,
...
 }
+
+////////////////////////////////////////////////////////////////////////////////
+/// This function checks if the timer is running within gSystem 
+/// (Has been started and did not finish yet).
+
+bool TTimer::IsRunning()
+{
+   if (gSystem && gSystem->GetListOfTimers())
+      return gSystem->GetListOfTimers()->IndexOf(this)!=-1;
```
Comment (pcanal) [inline core/base/src/TTimer.cxx:280]:
```suggestion
      return gSystem->GetListOfTimers()->IndexOf(this)! = -1;
```

File: core/base/src/TTimer.cxx [hunk 3]
```diff
@@ -269,3 +269,14 @@ void TTimer::SingleShot(Int_t milliSec, const char *receiver_class,
...
 }
+
+////////////////////////////////////////////////////////////////////////////////
+/// This function checks if the timer is running within gSystem 
+/// (Has been started and did not finish yet).
+
+bool TTimer::IsRunning()
+{
+   if (gSystem && gSystem->GetListOfTimers())
+      return gSystem->GetListOfTimers()->IndexOf(this)! = -1;
```
Comment (pcanal) [inline core/base/src/TTimer.cxx:280]:
```suggestion
      return gSystem->GetListOfTimers()->IndexOf(this)  != -1;
```

**Review by pcanal — APPROVED** [review]

**Comment by pcanal** [comment]
Thanks.

**Pull request merged**